FISH-6970 FISH-6973: App-specific OpenTelemetry configuration and integration

pdudits opened the pull request:
<!--- Title your PR with a Jira reference (if available) followed by brief description - for example: "PAYARA-1234 Add readme file" -->

## Description
<!-- Is this a fix or a feature? Does it address a GH issue? This section should be understandable by any developer without much background reading -->

An application can configure its OpenTelemetry SDK using microprofile config properties with prefix `otel.` using [AutoConfigure Facility](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure/README.md).
To enable the feature `otel.sdk.disabled=false` need to be set (as per MP spec).

In such case OpenTelemetry is enabled regardless of request tracing service state.
If Request Tracing is enabled and application requests opentelemetry, all traces are also exported to Request Tracing Service.
OpenTelemetry is enabled always when Request Tracing is enabled and uses standard same autoconfigure facility, using system properties and environment variables.

OpenTelemetry types are injectable as per MP spec.

Previous Tracer implementation is deprecated and it is not used anywhere now, can soon be removed.

OpenTelemetry and OpenTracing code converged into single module microprofile-telemetry. This merging will continue with next steps, where we ultimately aim to concentrate all of request tracing, opentracing and opentelemetry into few modules wherever distribution constraints allow for that.

## Important Info
### Blockers
<!--- Link any related or dependant PRs or issues here with brief description why -->

## Testing
### New tests
<!-- Link tests if they can be found in another repository or another PR -->
Passes 9/12 test cases of OpenTelemetry Tracing TCK. That includes REST endpoint instrumentation which is still done over opentracing shim (!). The failures however show that we'll need to redo that tracing in opentelemetry, because asserted trace metadata cannot be set via shim.

### Testing Performed
<!--- Please describe how you tested these changes. Which test suites did you run?  -->
MP OpenTracing passes, as well as propagation-related payara-samples

### Testing Environment
<!--- Which OS, JDK, Maven version did you use? - for example "Zulu JDK 11.0.11 on Ubuntu 18.04 with Maven 3.6.0"-->
```
Java version: 11.0.17, vendor: Azul Systems, Inc., runtime: C:\Program Files\Zulu\zulu-11
Default locale: en_US, platform encoding: Cp1252
OS name: "windows 11", version: "10.0", arch: "amd64", family: "windows"
```

## Documentation

Not yet, but there will be major changes, as support for custom opentracing tracers will also go away.

## Dependencies
payara/MicroProfile-TCK-Runners#212 makes OpenTracing TCK pass with this

pdudits commented:
Jenkins test please

pdudits commented:
Jenkins test please

pdudits commented:
Jenkins test please
```
The injected Tracer is required to be an instance of io.opentracing.mock.MockTracer but is instead
an instance of io.opentracing.mock.MockTracer@6ac6f51b
```
We need tracer to be in dependent scope for opentracing-mock to work.

Pandrex247 reviewed (changes requested):
I keep getting this error, is this something that we can control? We don't seem to have any documentation for OpenTelemetry over in our docs.
`[#|2023-02-13T12:28:54.883+0000|SEVERE|Payara 6.2023.3-SNAPSHOT|io.opentelemetry.exporter.internal.grpc.OkHttpGrpcExporter|_ThreadID=661;_ThreadName=OkHttp http://localhost:4317/...;_TimeMillis=1676291334883;_LevelValue=1000;|
  Failed to export spans. The request could not be executed. Full error message: Failed to connect to localhost/0:0:0:0:0:0:0:1:4317|#]`

Also you missed adding OpenTelemetry (and so by extension presumably OkHttp) to `3RD-PARTY-LICENSE.txt`

pdudits commented on appserver/payara-appserver-modules/pom.xml:
```diff
@@ -63,7 +63,6 @@
         <module>payara-rest-endpoints</module>
         <module>rest-monitoring</module>
         <module>cdi-auth-roles</module>
-        <module>opentracing-cdi</module>
```
The error means is that it has no opentelemetry endpoint to report to. It only happens when tracing is enabled, and we will advise to actually trace with external tool (or configure exporter to None).

I'll kill the remains of the module and have a look at license.

pdudits commented:
Added commits that finalize the implementation, fix cases where server was failing with tracing service turned on while also addressed your comments.

This version will not work without https://github.com/payara/MicroProfile-TCK-Runners/pull/212, because using external opentracing implementations is no longer possible.

arieki commented:
jenkins test please

arieki commented:
> jenkins test please

sorry I forgot MP TCK Runner haven't merged yett

Pandrex247 reviewed (approved):
I assume we'll have some documentation around this new log message (touch to configure, don't touch - bears):

`AutoConfiguredOpenTelemetrySdk found on classpath but automatic configuration is disabled. To enable, run your JVM with -Dotel.java.global-autoconfigure.enabled=true`

Other than that seems good!

Disclaimer: I haven't tested trying to use it with Jaeger or other OpenTracing impls we previously worked with

pdudits commented:
Yes, we'll have to flip a coin on what to do in this case, when something touches globally configured telemetry. We'll probably force it to NoOp now, so it remains silent. But let's wait until we've got the entire package in.

pdudits commented:
Jenkins test please

pdudits commented:
Jenkins test please

pdudits commented:
Jenkins test please

pdudits commented:
Jenkins test please

pdudits commented:
Jenkins test please
